Added missing supplementary material (#264 closes #535)

mjpost opened the pull request:
This adds missing attachments. I also had to change the schema because there is at least one frontmatter attachment. I haven't done much sanity checking.

mjpost commented:
I didn't realize that we had moved away from `<attachment type="X">` to `<X>` (e.g, `X=software`). I will have to update this.

mjpost commented:
Actually, it's not clear to me which is canonical. @mbollmann, can you direct here? Should I be using

    <software>...

or

    <attachment type="software">...

?

mbollmann commented:
Both work and do the same thing, since both were used in the XML when I wrote the code. I'm not aware that we've ever established a canonical version.

mbollmann commented:
That said I'd vote for `attachment` since it doesn't require us to add new XML tags when there's a new attachment type.

mjpost commented:
Okay, cool, I'll convert them all in this PR.

akoehn commented:
If you convert all, can you update the schema to disallow the other variant?

davidweichiang commented:
Agree, `attachment` is better.

knmnyn commented:
Agreed that `attachment` is better as well.  It'll make for much easier
processing downstream.


On Sun, Sep 29, 2019 at 3:24 AM David Chiang <notifications@github.com>
wrote:

> Agree, attachment is better.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/acl-org/acl-anthology/pull/536?email_source=notifications&email_token=AABU722JYCLYXWT3RIVX723QL6VNBA5CNFSM4I3IXBIKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD73AYYY#issuecomment-536218723>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABU7255BBPCTQGL622HAKLQL6VNBANCNFSM4I3IXBIA>
> .
>

mjpost commented:
Done.

mbollmann commented on data/xml/P14.xml:
```diff
@@ -2633,7 +2654,8 @@
       <doi>10.3115/v1/P14-2115</doi>
-      <dataset>P14-2115.Datasets.tgz</dataset>
+      <attachment type="dataset">P14-2115.Datasets.tgz</attachment>
+      <attachment type="datasets">P14-2115.Datasets.tgz</attachment>
```
Seems that this attachment type is added twice?

mjpost replied:
Oh, drat. I checked for missing attachments before the tag conversion.

mbollmann commented on data/xml/W19.xml:
```diff
@@ -2126,9 +2129,10 @@
+      <attachment type="software">W19-2009.Software.zip</attachment>
       <attachment type="poster">W19-2009.Poster.pdf</attachment>
       <doi>10.18653/v1/W19-2009</doi>
+      <attachment type="software">W19-2009.Software.zip</attachment>
```
Actually, this is too...

mjpost commented:
Anyone have a minute to check this over? It should be pretty non-controversial—I just

- converted <dataset>, <software>, and <presentation> tags to <attachment>
- added attachments that were on the server but were not in the XML
- added in the script I used to do it

I verified by script that there are no duplicate attachments.

mjpost commented:
Ping @acl-org/acl-style @mbollmann @aryamccarthy @davidweichiang @akoehn
